tools/doc/man.js

```diff
@@ -0,0 +1,133 @@
+// Copyright AJ Jordan and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+'use strict';
+
+const assert = require('assert');
+const path = require('path');
+const remark = require('remark');
+const man = require('remark-man');
+const parser = remark().use(fixupHeader).use(synopsis).use(stability).use(description).use(man);
+const dontBuild = ['_toc', 'addons', 'cli', 'deprecations', 'documentation', 'domain', 'errors', 'globals', 'index', 'intl', 'process', 'punycode', 'synopsis', 'tracing', 'v8'];
+
+module.exports = toMan;
+
+function fixupHeader(opts) {
+  return function _fixupHeader(ast, file, next) {
+    const header = ast.children[0];
+
+    assert.strictEqual(header.type, 'heading', 'First node is not a header');
+    assert.strictEqual(header.depth, 1, 'Header is not of depth 1');
+
+    const moduleName = header.children[0].value;
+    header.children[0].value = `node-${moduleName.toLowerCase()}(3)`;
+    header.children[0].value += ` -- Node.js ${moduleName} module`;
+
+    // Attach the module name for other plugins in this file to use
+    file.moduleName = moduleName;
+
+    next();
+  };
+}
+
+function synopsis(opts) {
+  return function _synopsis(ast, file, next) {
+    const moduleName = file.moduleName.toLowerCase();
+
+    const heading = {
```

Comment:
This way of building nodes is kinda fugly but I'm not sure there's a better way. https://github.com/eush77/unist-builder perhaps?